blueflood-http/src/main/java/com/rackspacecloud/blueflood/inputs/handlers/HttpMetricsIngestionHandler.java

```diff
@@ -84,12 +93,31 @@ public HttpMetricsIngestionHandler(HttpMetricsIngestionServer.Processor processo
     }
 
     protected JSONMetricsContainer createContainer(String body, String tenantId) throws JsonParseException, JsonMappingException, IOException {
+        //mapping
         List<JSONMetric> jsonMetrics =
                 mapper.readValue(
                         body,
                         typeFactory.constructCollectionType(List.class, JSONMetric.class)
                 );
-        return new JSONMetricsContainer(tenantId, jsonMetrics);
+
+        //validation
+        List<ErrorResponse.ErrorData> validationErrors = new ArrayList<ErrorResponse.ErrorData>();
+        List<JSONMetric> validJsonMetrics = new ArrayList<JSONMetric>();
+
+        for (JSONMetric metric: jsonMetrics) {
+            Set<ConstraintViolation<JSONMetric>> constraintViolations = validator.validate(metric);
+
+            if (constraintViolations.size() == 0) {
+                validJsonMetrics.add(metric);
+            } else {
+                for (ConstraintViolation<JSONMetric> constraintViolation : constraintViolations) {
+                    validationErrors.add(new ErrorResponse.ErrorData(tenantId, metric.getMetricName(),
+                            constraintViolation.getPropertyPath().toString(), constraintViolation.getMessage()));
```

Comment:
Looks like the source is set to be the variable name.  Should we use something else that won't be mistakenly changed by a later developer?  I.e., define a static string to be the signifier.  Can that be set in the constraintViolation?